docs/docs/etl-pipeline-tutorial/2-add-a-resource.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Add a resource
+description: Add a resource to your assets
+sidebar_position: 30
+---
+
+We've now created our own assets and combined them with assets from a component. We will circle back to the ingestion assets we defined and include another Dagster object to help how we manage our database connection with DuckDB.
+
+In this step, you will:
+
+- Add a DuckDB resource
+
+## 1. Define the DuckDB resource
+
+In Dagster, [resources](/guides/build/external-resources) are reusable components that provide external context or functionality such as  database connections, clients, or configurations. These can be used by a number of different Dagster objects but we will first apply them to our assets.
+
+We will use the `dagster-duckdb` library so add it to our project:
+
+```bash
+uv pip install dagster-duckdb pandas
+```
+
+Next, we need to scaffold our resources in the same way as our assets using `dg`:
+
+```bash
+dg scaffold defs dagster.resources resources.py
+```
+
+This adds a file, `resources.py`, to the `etl_tutorial` module:
+
+<CliInvocationExample path="docs_snippets/docs_snippets/guides/tutorials/etl_tutorial/tree/resources.txt" />
+
+Within this file we will define the resource using the <PyObject section="definitions" module="dagster" object="Definitions" decorator />.
+
+<CodeExample
+  path="docs_snippets/docs_snippets/guides/tutorials/etl_tutorial/src/etl_tutorial/defs/resources.py"
+  language="python"
+  title="src/etl_tutorial/defs/resources.py"
+/>
+
+This centralizes our connection to DuckDB into a single object. Currently each of our assets handles that connection separately. Now the connection can be shared across all our assets.
+
+```mermaid
+%%{
+  init: {
+    'theme': 'base',
+    'themeVariables': {
+      'primaryColor': '#4F43DD',
+      'primaryTextColor': '#FFFFFF',
+      'primaryBorderColor': '#231F1B',
+      'lineColor': '#DEDDFF',
+      'secondaryColor': '#BDBAB7',
+      'tertiaryColor': '#FFFFFF'
+    }
+  }
+}%%
+  graph LR
+    style DuckDB fill:#BDBAB7,stroke:#BDBAB7,stroke-width:2px
+
+    Asset -.-> Resource
+    Resource -.-> DuckDB
+```
+
+## 2. Add a resource to our assets
+
+With our resource defined we need to update our asset code. Since all of our ingestion assets rely on the `import_url_to_duckdb` to execute the query we will first update that function to use the `DuckDBResource` to handle query execution:
```

Comment:
```suggestion
With our resource defined, we need to update our asset code. Since all of our ingestion assets rely on the `import_url_to_duckdb` to execute the query, we will first update that function to use the `DuckDBResource` to handle query execution:
```